src/App.tsx

```diff
@@ -34,170 +70,79 @@ export const App: React.FC = () => {
-          </span>
-
-          {/* Active link should have the 'selected' class */}
-          <nav className="filter" data-cy="Filter">
-            <a
-              href="#/"
-              className="filter__link selected"
-              data-cy="FilterLinkAll"
-            >
-              All
-            </a>
-
-            <a
-              href="#/active"
-              className="filter__link"
-              data-cy="FilterLinkActive"
-            >
-              Active
-            </a>
+        {todoos.length && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {numberOfActiveTodos} items left
+            </span>
 
-            <a
-              href="#/completed"
-              className="filter__link"
-              data-cy="FilterLinkCompleted"
+            {/* Active link should have the 'selected' class */}
+            <nav className="filter" data-cy="Filter">
+              <a
+                href="#/"
+                className={cn('filter__link', {
+                  selected: filterOption === FilterOptions.All,
+                })}
+                data-cy="FilterLinkAll"
+                onClick={() => setFilterOption(FilterOptions.All)}
+              >
+                All
+              </a>
+
+              <a
+                href="#/active"
+                className={cn('filter__link', {
+                  selected: filterOption === FilterOptions.Active,
+                })}
+                data-cy="FilterLinkActive"
+                onClick={() => setFilterOption(FilterOptions.Active)}
+              >
+                Active
+              </a>
+
+              <a
+                href="#/completed"
+                className={cn('filter__link', {
+                  selected: filterOption === FilterOptions.Completed,
+                })}
+                data-cy="FilterLinkCompleted"
+                onClick={() => setFilterOption(FilterOptions.Completed)}
+              >
+                Completed
+              </a>
+            </nav>
+
+            {/* this button should be disabled if there are no completed todos */}
+            <button
+              type="button"
+              className="todoapp__clear-completed"
+              data-cy="ClearCompletedButton"
             >
-              Completed
-            </a>
-          </nav>
-
-          {/* this button should be disabled if there are no completed todos */}
-          <button
-            type="button"
-            className="todoapp__clear-completed"
-            data-cy="ClearCompletedButton"
-          >
-            Clear completed
-          </button>
-        </footer>
+              Clear completed
+            </button>
+          </footer>
+        )}
       </div>
 
       {/* DON'T use conditional rendering to hide the notification */}
       {/* Add the 'hidden' class to hide the message smoothly */}
       <div
```

Comment:
I would recommend taking more components out separately, such as Header, Footer and ErrorNotification, as they are quite complex and in theory could be overused.